fixtures/xunit_tools.py

```diff
@@ -0,0 +1,253 @@
+                else:
+                    f.write("{}\n".format(test))
+
+    test_name = []
+
+    testcases = etree.Element("testcases")
+    testcases.attrib['project-id'] = xunit['project_id']
+    response_properties = etree.Element("response-properties")
+    response_property = etree.Element(
+        "response-property", name=xunit['response']['id'], value=xunit['response']['value'])
+    response_properties.append(response_property)
+    properties = etree.Element("properties")
+    lookup = etree.Element("property", name="lookup-method", value="custom")
+    properties.append(lookup)
+    testcases.append(response_properties)
+    testcases.append(properties)
+
+    for item in items:
+        work_items = []
+        custom_fields = {}
+        try:
+            description = item.function.func_doc
+        except:
+            pass
+        try:
+            requirement = item.get_marker('requirement').args[0]
+            from utils.conf import cfme_data
+            requirement_id = cfme_data['requirements'][requirement]
+            work_items.append({'id': requirement_id, 'role': 'verifies'})
+        except:
+            pass
+        try:
+            tier = item.get_marker('tier').args[0]
+            tier_id = caselevels[str(tier)]
+            custom_fields['caselevel'] = tier_id
+        except:
+            pass
+        try:
+            params = item.callspec.params
+            param_list = params.keys()
+        except:
+            param_list = []
+        try:
+            automated = item.get_marker('automated').args[0]
+        except:
+            automated = True
+        if automated:
+            # The master here should probably link the latest "commit" eventually
+            automation_script = 'http://github.com/{}/{}/blob/master/{}#L{}'.format(
+                xunit['gh_owner'],
+                xunit['gh_repo'],
+                item.location[0],
+                item.function.func_code.co_firstlineno
+            )
+            custom_fields['caseautomation'] = "automated"
+            custom_fields['automation_script'] = automation_script
+        description = '{}<br/><br/><a href="{}">Test Source</a>'.format(
+            description, automation_script)
+
+        name = item.name
+        name = re.sub('\[.*\]', '', name)
+        if name not in test_name:
+            test_name.append(name)
+            testcases.append(
+                testcase_gen(
+                    name,
+                    description=description,
+                    parameters=param_list,
+                    linked_items=work_items,
+                    custom_fields=custom_fields))
+
+    xml = etree.ElementTree(testcases)
+    xml.write('test_case_import.xml', pretty_print=True)
+
+    tests = []
+    for item in items:
+        name = re.sub('\[.*\]', '', item.name)
+        try:
+            params = item.callspec.params
+            param_dict = {p: get_name(v) for p, v in params.iteritems()}
+        except:
```

Comment:
- To catch any exception use the ```except Exception```, instead of ```except BaseException``` or just ```except```.
- Indicate the most specific exception type in ```except``` block.

For example:
```python
# Bad
try:
    mapping[key]
except Exception:
    ...

# Better
try:
    mapping[key]
except KeyError:
    ...
```